lib/project/setup-project.js

```diff
@@ -2,32 +2,65 @@
 const semverCoerce = require('semver/functions/coerce');
 const YarnLockfile = require('@yarnpkg/lockfile');
 const npa = require('npm-package-arg');
+const ini = require('ini');
 const fs = require('fs');
 const path = require('path');
 const debug = require('debug')('supported:project');
+const { setupProjectPath } = require('../read-from-url');
 
 const npmConfig = require('../npm/config');
 
-module.exports = async function setupProject(projectRoot) {
-  const config = await npmConfig(projectRoot); // kinda slow, TODO: re-implement as standalone lib
-  // const { policies } = options;
-  const pkgPath = `${projectRoot}/package.json`;
-  if (!fs.existsSync(pkgPath)) {
-    throw new Error(`${pkgPath} does not exist, are you sure this is a valid package?`);
-  }
-  if (!fs.statSync(pkgPath).isFile()) {
-    throw new Error(`${pkgPath} is not a file, are you sure this is a valid package?`);
+module.exports = async function setupProject(projectRoot, flags = {}) {
+  let projectInfo = {};
+  if (!fs.existsSync(projectRoot)) {
+    projectInfo = await setupProjectPath(projectRoot, {
+      token: flags.token,
+      hostUrl: flags.hostUrl,
+    });
+    // if project root is a URL then use homedir as root
+    projectRoot = require('os').homedir();
```

Comment:
This is surprising, I understanding reading the `npmrc` from the the homedir, but what if someone has a spurious lockfile or package.json in there homedir? This happens surprisingly often, and leads to a confusing and potentially tricky problem too debug.

Is there a more resilient alternative?

Reply:
We are not using lock file or package.json from the homedir even if it exists. those are all read from in-memory files passed. The only npmrc we read is from homedir to avoid possible settings in the current project. The hope is not to have any custom npmrc in homedir.